### PR TITLE
Better PropType validation for data in AbstractSeries; Tests for using Accessors

### LINKED ISSUE #724: PropType warning of data format supplied to LineSeries
I am able to supply a 2d array as `data` to LineSeries and render it with accessors `getX: d => d[0]` and `getY: d => d[1]`. However I still get an error in the console: Failed prop type: Invalid prop `data[0]` of type `array` supplied to `LineSeries`, expected `object`.

### PR DESCRIPTION
**Pull request by manassra**
Fixes #724. This change will prevent prop-validation for throwing a warning when users use `data` accessors (for using arrays, for example).

### CODE REVIEW

File: src/plot/series/abstract-series.js [hunk 1]
```diff
@@ -39,7 +39,10 @@ const propTypes = {
   ...getScalePropTypesByAttribute('color'),
   width: PropTypes.number,
   height: PropTypes.number,
-  data: PropTypes.arrayOf(PropTypes.object),
+  data: PropTypes.arrayOf(PropTypes.oneOfType([
```
Comment (mcnuttandrew) [inline src/plot/series/abstract-series.js]: add some documentation/testing to show that this case is supported?

Comment (manassra) [reply src/plot/series/abstract-series.js]: you mean testing that the warning is not triggered anymore? I did test this manually, but I am not sure how to achieve this through automated tests.

Comment (mcnuttandrew) [reply src/plot/series/abstract-series.js]: more like add a test case demonstrating that this is a valid way to input data (I don't remember writing any but I might have?) and add a note in the documentation showing that this is a supported case (again I may have already done that but i don't think i did??). If it's the case that I am being dumb and these things already exist than this gets a stamp

File: docs/xy-plot.md [hunk 2]
```diff
@@ -54,6 +54,24 @@ XYPlot is a wrapper for series, hints, axes and other components. Most of these
...
 * `style` (optional) - css properties as an object.
 
+Accessors can also be used to retreieve the properties above from the `data` object. For instance, the `getX` and `getY` accessors can be passed to the XYPlot object to access the `x` and `y` properties from `data` for each series. 
+
+```jsx
+<XYPlot
+  width={300}
+  height={300}
+  getX={d => d[0]}
+  getY={d => d[1]>
```
Comment (schnerd) [inline docs/xy-plot.md]: I know that this PR isn't implementing accessors, but now I'm curious–do we also support accessors on series components? That seems like a more common use case that people might have (and if we support it we should document it).

```
const data = [
  {fruit: 'apple', bought: 23, sold: 18},
  {fruit: 'banana', bought: 28, sold: 27}
];

<XYPlot
   ...
   getX{d => d.fruit}
/>
   <LineSeries getY={d => d.bought} />
   <LineSeries getY={d => d.sold} />
</XYPlot>
```

@jckr @mcnuttandrew

Comment (mcnuttandrew) [reply docs/xy-plot.md]: Series level accessors *might* be supported, though if they are the functionality is untested. The only guaranteed support that we offer is on a plot level. I'd support adding accessors on series, seems like a reasonable idea.

Comment (manassra) [reply docs/xy-plot.md]: I don't think they are supported. the accessors seem to be passed down from `XYPlot`.

**Pull request merged**